node/router/src/lib.rs

```diff
@@ -83,8 +83,20 @@ const PEER_CACHE_FILENAME: &str = "cached_router_peers";
 pub trait PeerPoolHandling<N: Network>: P2P {
     const OWNER: &str;
 
+    /// The maximum number of peers permitted to be stored in the peer pool.
+    const MAXIMUM_POOL_SIZE: usize = 1_000;
```

Comment:
I would not set any defaults so we have uniform reading experience for gateway v.s. router.

An individual peer may send us 256 ip addresses, so let's stick to 10k so we don't go over the limit so quickly.